apps/ftu/js/wifi.js

```diff
@@ -230,8 +231,10 @@ var WifiUI = {
     // Retrieve SSID from dataset
     var ssid = event.target.dataset.ssid;
 
+    var selectedNetwork = WifiManager.getNetwork(ssid);
+
     // Do we need to type password?
-    if (WifiHelper.isOpen(WifiManager.getNetwork(ssid))) {
+    if (WifiHelper.isOpen(selectedNetwork)) {
```

Comment:
maybe do if (selectedNetwork && isOpen(selectedNetwork) ) ? We get undefined behavior is isOpen is passed null or undefined